### PR TITLE
Added more unit testing for AgentPluginManager

### PR DESCRIPTION
**Pull request by jtpetty**

### CODE REVIEW

File: src/Agent.Worker/AgentPluginManager.cs [hunk 1]
```diff
@@ -122,6 +103,29 @@ public async Task RunPluginTaskAsync(IExecutionContext context, string plugin, D
...
+
+            // Resolve the working directory.
+            string workingDirectory = HostContext.GetDirectory(WellKnownDirectory.Work);
+            ArgUtil.Directory(workingDirectory, nameof(workingDirectory));
+
+            // Agent.PluginHost
+            string file = Path.Combine(HostContext.GetDirectory(WellKnownDirectory.Bin), $"Agent.PluginHost{Util.IOUtil.ExeExtension}");
+            ArgUtil.File(file, $"Agent.PluginHost{Util.IOUtil.ExeExtension}");
+
+            var pluginContext = GeneratePluginExecutionContext(context, inputs, runtimeVariables);
```
Comment (damccorm) [inline src/Agent.Worker/AgentPluginManager.cs]: Is this used anywhere? I'm a little confused what you're doing here

File: src/Agent.Worker/AgentPluginManager.cs [hunk 2]
```diff
@@ -133,6 +137,8 @@ public async Task RunPluginTaskAsync(IExecutionContext context, string plugin, D
                 // Execute the process. Exit code 0 should always be returned.
                 // A non-zero exit code indicates infrastructural failure.
                 // Task failure should be communicated over STDOUT using ## commands.
+                // Agent.PluginHost's arguments
+                string arguments = $"task \"{plugin}\"";
```
Comment (damccorm) [inline src/Agent.Worker/AgentPluginManager.cs]: Any reason this was moved? Not really a problem, but it does make the comments a little confusing.

File: src/Agent.Worker/AgentPluginManager.cs (hunk 1 above)
Comment (jtpetty) [reply src/Agent.Worker/AgentPluginManager.cs]: I wanted to write a unit test for this logic, but not necessarily kick off the execution of the plugin process so I pulled it out into its own method.

File: src/Agent.Worker/AgentPluginManager.cs (hunk 2 above)
Comment (jtpetty) [reply src/Agent.Worker/AgentPluginManager.cs]: I think I just moved it because it was defined so far away from where it was being used.

Comment (jtpetty) [reply src/Agent.Worker/AgentPluginManager.cs]: Oh, I guess I should have put an extra line between that existing comment block and the relocated, unrelated, unnecessary comment that went along with the variable declaration.

File: src/Agent.Worker/AgentPluginManager.cs (hunk 1 above)
Comment (damccorm) [reply src/Agent.Worker/AgentPluginManager.cs]: Ah, I see - this diff was killing me.

**Pull request merged**